cni/network/invoker_azure.go

```diff
@@ -0,0 +1,99 @@
+package network
+
+import (
+	"net"
+
+	"github.com/Azure/azure-container-networking/cni"
+	"github.com/Azure/azure-container-networking/common"
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/network"
+	cniSkel "github.com/containernetworking/cni/pkg/skel"
+	cniTypesCurr "github.com/containernetworking/cni/pkg/types/current"
+)
+
+type AzureInvoker struct {
+	plugin *netPlugin
+}
+
+func NewAzureInvoker(plugin *netPlugin) *AzureInvoker {
+	return &AzureInvoker{
+		plugin: plugin,
+	}
+}
+
+func (invoker *AzureInvoker) Add(args *cniSkel.CmdArgs, nwCfg *cni.NetworkConfig, nwInfo network.NetworkInfo, isDeletePoolOnError bool) (*cniTypesCurr.Result, *cniTypesCurr.Result, error) {
+	var (
+		result   *cniTypesCurr.Result
+		resultV6 *cniTypesCurr.Result
+		err      error
+	)
+
+	if len(nwInfo.Subnets) > 0 {
+		nwCfg.Ipam.Subnet = nwInfo.Subnets[0].Prefix.String()
+	}
+
+	// Call into IPAM plugin to allocate an address pool for the network.
+	result, err = invoker.plugin.DelegateAdd(nwCfg.Ipam.Type, nwCfg)
+	if err != nil {
+		err = invoker.plugin.Errorf("Failed to allocate pool: %v", err)
+		return result, resultV6, err
+	}
+
+	defer func() {
+		if err != nil {
+			invoker.plugin.ipamInvoker.Delete(result.IPs[0].Address, nwCfg, nwInfo, isDeletePoolOnError)
+		}
+	}()
+
+	if nwCfg.IPV6Mode != "" {
+		nwCfg6 := nwCfg
+		nwCfg6.Ipam.Environment = common.OptEnvironmentIPv6NodeIpam
+		nwCfg6.Ipam.Type = ipamV6
+
+		if len(nwInfo.Subnets) > 1 {
+			nwCfg6.Ipam.Subnet = nwInfo.Subnets[1].Prefix.String()
```

Comment:
magic number index 1. Know this is old code you're just moving.

Do we have some backlog item to try and prove ipv6 works with cns?